core/modules/system/system.admin.inc

```diff
@@ -602,10 +604,16 @@ function system_modules($form, $form_state = array()) {
       $extra['required_by'][] = $module->info['explanation'];
     }
 
-    // Check to see if this is a core module.
+    // Check to see if this is a core/contrib/custom.
     if (strstr($module->uri, 'core/modules')) {
       $extra['core'] = TRUE;
     }
+    elseif (!$module->info['project'] && !$module->info['version'] && !$module->info['timestamp']) {
```

Comment:
There are plenty of core modules that have both project and version. Perhaps we should test for timestamp alone? (Anding these conditions together would return none of my custom modules.)